### ISSUE TITLE
As an admin, I want changes made to authors and books in the admin interface to automatically update the book search, so that content in the search and admin interface stay in sync.

### ISSUE THREAD

**Issue by rlskoeser** [Princeton-CDH/mep-django#425]
# testing notes
When you edit records in the django admin, they should be automatically updated in the Solr index so that data edits are reflected in the public site.  Test by making changes in the django admin and then reloading the book search/browse page to see that your changes have been propagated.

- books
   - edit fields on the work record (e.g. title or publication date)
   - edit the person record for an author (e.g. sort name)
   - change author associated with a work:
      - add a new author
      - remove an author
       - change the person linked as author
- edit creator type: rename Author to something else (this will take a while, since it will reindex all books). In the list display no books should have authors listed. (Not something we expect to do in practice, but other creator types may change)

Split out from #339

# dev notes
- [x] creators
- [x] person
- [x] creatortype

**Comment by clmahoney** [comment]
- YES, if I edit title in the work record, it changes on the site. I changed " 'Twixt Land and Sea" to " 'Twixt You and Me" and it shows up like that :) Then changed it back
- also YES, if I change the pub date, it changes on the site. Changed the above work to 1912 to 1911
- YES, if I edit the person record for an author (e.g. sort name) it changes - tried with "Conrad, Joseph" - changed it to "Conradi, Joseph" and that was mirrored in the site!
- YES, I can change author associated with a work: I added Theodore Dreiser to Ben Hecht's "1001 Afternoons in Chicago" and sure enough, he showed up along with Hecht
- YES, I can delete an author too - I deleted Dreiser and he vanished from the site
- YES, I changed Ben Hecht from author to editor of "1001 Afternoons in Chicago" and that was mirrored on the site
--> this testing also revealed an error in the book work, which is that Hecht's book is published earlier than was listed and should be written out "One Thousand and One Afternoons in Chicago," which I fixed!

I'm confused about this one: I can rename author to something else like editor (which I did above), but author names ARE listed in the list view on the site. If I change it from author to something else, that doesn't show up. Just author.
